From 60930805393a649f1df0f887f889b7c23d9d4d3a Mon Sep 17 00:00:00 2001
From: poly000 <1348292515@qq.com>
Date: Thu, 20 Jul 2023 06:30:38 +0800
Subject: [PATCH] feat #2056: add `xesam:url` field for MPRIS

chore: do not use fuo scheme, only netease music id is preserved

fix: lint prettier error
---
 src/electron/mpris.js | 1 +
 src/utils/Player.js   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/electron/mpris.js b/src/electron/mpris.js
index 2ea5073..f1272d7 100644
--- a/src/electron/mpris.js
+++ b/src/electron/mpris.js
@@ -35,6 +35,7 @@ export function createMpris(window) {
       'xesam:title': metadata.title,
       'xesam:album': metadata.album,
       'xesam:artist': metadata.artist.split(','),
+      'xesam:url': metadata.url,
     };
   });
 
diff --git a/src/utils/Player.js b/src/utils/Player.js
index 21c90f6..f4d6fb9 100644
--- a/src/utils/Player.js
+++ b/src/utils/Player.js
@@ -612,6 +612,7 @@ export default class {
       ],
       length: this.currentTrackDuration,
       trackId: this.current,
+      url: '/trackid/' + track.id,
     };
 
     navigator.mediaSession.metadata = new window.MediaMetadata(metadata);
-- 
2.41.0

